jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -239,41 +239,49 @@ public StepsRunner pushImage() {
   }
 
   public StepsRunner loadDocker(DockerClient dockerClient) {
-    createRootProgressAllocation("Build to Docker daemon");
+    rootProgressAllocationDescription = "Build to Docker daemon";
 
     return enqueueStep(
         () ->
             steps.finalStep =
                 new LoadDockerStep(
                     listeningExecutorService,
                     buildConfiguration,
-                    Preconditions.checkNotNull(rootProgressAllocation),
+                    Preconditions.checkNotNull(rootProgressEventDispatcher).newChildProducer(),
                     dockerClient,
                     Preconditions.checkNotNull(steps.pullAndCacheBaseImageLayersStep),
                     Preconditions.checkNotNull(steps.buildAndCacheApplicationLayerSteps),
                     Preconditions.checkNotNull(steps.buildImageStep)));
   }
 
   public StepsRunner writeTarFile(Path outputPath) {
-    createRootProgressAllocation("Build to tar file");
+    rootProgressAllocationDescription = "Build to tar file";
 
     return enqueueStep(
         () ->
             steps.finalStep =
                 new WriteTarFileStep(
                     listeningExecutorService,
                     buildConfiguration,
-                    Preconditions.checkNotNull(rootProgressAllocation),
+                    Preconditions.checkNotNull(rootProgressEventDispatcher).newChildProducer(),
                     outputPath,
                     Preconditions.checkNotNull(steps.pullAndCacheBaseImageLayersStep),
                     Preconditions.checkNotNull(steps.buildAndCacheApplicationLayerSteps),
                     Preconditions.checkNotNull(steps.buildImageStep)));
   }
 
   public BuildResult run() throws ExecutionException, InterruptedException {
-    Preconditions.checkNotNull(rootProgressAllocation);
-    stepsRunnable.run();
-    return Preconditions.checkNotNull(steps.finalStep).getFuture().get();
+    Preconditions.checkNotNull(rootProgressAllocationDescription);
+
+    try (ProgressEventDispatcher progressEventDispatcher =
```

Comment:
nit: `try (rootProgressEventDispatcher) = ProgressEventDispatcher.newRoot(...)) {`

Reply:
`try-with-resource` requires creation of a new local variable.